### ISSUE TITLE
Why softmax for binary cross entropy loss

### ISSUE THREAD

**Issue by creative-swamy** [AyanKumarBhunia/Deep-One-Shot-Logo-Retrieval#7]
1. As it is mentioned in the paper, how relevant it is to use soft max for binary cross entropy loss.? Kindly explain. 
2. tf.keras.losses.bianry_crossentropy averages the loss only in the W(Width) direction. In the paper, it's mentioned to take average in H(Height) direction also. Please clarify if my understanding is correct or not. 
3. Having softmax as the final layer does not show any good results. With sigmoid as the final layer the validation accuracy is reaching 98% with the validation loss of 3%. Please clarify what layer have been used in your experiments.
4. For training I have generated 154560 triplets and 90% of that has been used for training and remaining for validation. Please provide accuracy and loss statistics of your experiments.

**Comment by creative-swamy** [comment]
![refpic](https://user-images.githubusercontent.com/43458445/115946285-2cde2780-a48e-11eb-86c4-891a2bf30382.png)

**Comment by AyanKumarBhunia** [comment]
@creative-swamy You need to use sigmoid instead of softmax.  

'SOFTMAX' seems to be a typo in the paper. Some people also raised that issue -- we should have uploaded a more recent version in the arxiv -- though we can't change the PR version. 

If you apply -- sigmoid -- the output tensor dimension will be  Batch_Size, H, W, 1 -- with Sigmoid.

I hope that clears your doubts. This project is almost 3/4 years old -- now I don't have those log files. So, I can't help you with the loss statistics at this moment.

**Comment by creative-swamy** [comment]
Thank you for the quick response and that helps me a lot.

**Comment by creative-swamy** [comment]
Great work!
